src/openrct2/windows/ride.c

```diff
@@ -1834,7 +1834,9 @@ static void window_ride_init_viewport(rct_window *w)
 		rct_ride_entry* ride_entry = get_ride_entry_by_ride(ride);
 		if (ride_entry && ride_entry->tab_vehicle != 0){
 			rct_vehicle* vehicle = GET_VEHICLE(focus.sprite.sprite_id);
-			focus.sprite.sprite_id = vehicle->next_vehicle_on_train;
+			if (vehicle->next_vehicle_on_train < countof(ride_entry->vehicles)) {
```

Comment:
If there's a chance of a valid sprite identifier that's not a vehicle (rather than just the trivially invalid `0xFFFF`), then absolutely. I'll have it do that. Did not realise that about `ride_entry->vehicles` - naively thought it'd be the count of vehicles a given ride has.